src/components/WebexRemoteMedia/WebexRemoteMedia.js

```diff
@@ -17,15 +16,41 @@ import './WebexRemoteMedia.scss';
  * NOTE: waiting for the UX for a design on what to display if there is no remote video
  */
 export default function WebexRemoteMedia({className, meetingID}) {
-  const {remoteAudio, remoteVideo, error} = useMeeting(meetingID);
+  const {remoteAudio, remoteVideo, remoteShare, error} = useMeeting(meetingID);
+  const shareControl = useShareControl(meetingID);
   const audioRef = useStream(remoteAudio);
   const videoRef = useStream(remoteVideo);
-  const hasMedia = !!(remoteAudio || remoteVideo);
+  const shareRef = useStream(remoteShare);
+  const hasMedia = !!(remoteAudio || remoteVideo || remoteShare);
   const mainClasses = {
     [`${WEBEX_COMPONENTS_CLASS_PREFIX}-remote-media`]: true,
     [className]: !!className,
   };
 
+  const isLocalSharingInactive = shareControl.state === 'inactive';
+
+  //
+  // Switch the visibility of the remoteShare video by checking its readyState
+  // Note: Sharing detection can be removed from component, once the sdk provides
+  //       the Start/Stop events about the remote sharing.
+  //
+  const detectRemoteSharing = () => {
```

Comment:
This is not necessary in this component. `WebexMeeting` component handles the layering of local media + remote media + meeting controls. `WebexRemoteMedia` should only be concerned with displaying remote streams.
Do you have suggestions on how to make that clearer in the docs/repo?

Reply:
This is the tricky part. 

The idea of `if (remoteShare) then display <video>` doesn't work for the sdk, because the remoteShare/remoteVideo stream object will be always true value during the entire meeting. The difference is that it receives data or not, and the stream objects were always there.

And, unfortunately, the sdk doesn't provide the remote start/stop events as well, so the adapter can't provide the states of those remote media. That means UI components have no way to know the remote start/stop sharing and cannot do proper UI changes.

This is a sdk design problem plus a missing sdk feature.

But we still found a way to detect the remote event from the DOM video element as a workaround, that's detectRemoteSharing(). DOM element was managed by React, so the code exists in the component.

We can surely remove the detection function, but the UI will lose ability to provide feedbacks to the users about the changes of remote steam (with the current sdk).

Reply:
A couple things here:
* The JSON adapter code seems to be making changes to `localShare` only, so why is the remote media component being changed?
* I think there might be some rework that needs to happen on the SDK adapter if what you're saying is how things are currently done. SDK issues/design should not be bubbling up to the components, thats why the adapter is there. The SDK adapter should do what it needs to to make sure that if there is something being shared, then it returns the right stream, whereas if nothing is being shared, then the attribute should be `null`. This is why in the SDK adapter there are two objects `meeting` which is the component systems state, and `sdkMeeting`, which is the state of how the SDK handles meetings; the job of the adapter to to translate `sdkMeeting` into `meeting` so that the components don't have to know any of those implementation details.